var/spack/repos/builtin.mock/packages/cmake-client/package.py

```diff
@@ -109,7 +110,7 @@ def install(self, spec, prefix):
         print(cmake)
         print(cmake.exe)
         check(
-            cmake.exe[0].startswith(spec["cmake"].prefix.bin),
+            str(PurePath(cmake.exe[0])).startswith(spec["cmake"].prefix.bin),
```

Comment:
What is the effect of this? I can't tell if it's because `spec...prefix` is POSIX, or if `cmake.exe` is posix. I wonder if later adjustments to `Spec.prefix` would cause an issue here, and if we should wrap both entities with `PurePath` (with a TODO comment to address `Spec.prefix`).